mongo-utils/src/main/java/org/wso2/mongo/MongoDBDataSourceUtil.java

```diff
@@ -64,18 +64,19 @@ private MongoDBDataSourceUtil() {
     }
 
     public static Object initClient(BMap<BString, BValue> config) {
-        String host = config.getStringValue(fromString("host")).getValue();
-        long port = config.getIntValue(fromString("port"));
+        String host = config.containsKey(fromString("host")) ? config.getStringValue(fromString("host")).getValue() 
+                      : "";
+        long port = config.containsKey(fromString("port")) ? config.getIntValue(fromString("port")) : 0;
         String username = "";
         // Optional Fields
-        if (config.getStringValue(fromString("username")) != null) {
+        if (config.containsKey(fromString("username"))) {
             username = config.getStringValue(fromString("username")).getValue();
         }
         String password = "";
-        if (config.getStringValue(fromString("password")) != null) {
+        if (config.containsKey(fromString("password"))) {
             password = config.getStringValue(fromString("password")).getValue();
         }
-        BMap options = config.getMapValue(fromString("options"));
+        BMap options = config.containsKey(fromString("options")) ? config.getMapValue(fromString("options")) : null;
 
         try {
             return init(host, port, username, password, options);
```

Comment:
I guess username and password is mandatory for initialisation. If they are not passed, init will fail and give a meaningful error? It can happen upon wrong username password as well.

Reply:
Username and Password are not mandatory. There are some alternative methods to initialize without username, password. Giving connection URL in the options of the config is one of such methods.  In such scenarios, they don't need to pass in init config. How ever, it will throw a meaningful error when it doesn't have at-least any of the authentication mechanism. It will throw error upon wrong username password as well.